### PR TITLE
[MOE] changed parallelmode to dist process group

### PR DESCRIPTION
**Pull request by 1SAA**

### CODE REVIEW

File: colossalai/nn/layer/moe/_operation.py [hunk 1]
```diff
@@ -18,35 +15,35 @@
 class AllGather(torch.autograd.Function):
 
     @staticmethod
-    def forward(ctx: Any, inputs: Tensor, parallel_mode: ParallelMode) -> Tensor:
+    def forward(ctx: Any, inputs: Tensor, group=None) -> Tensor:
```
Comment (feifeibear) [inline colossalai/nn/layer/moe/_operation.py:18]: group : Optional[ParallelGroup] = None?
you missed the typing hit here

File: colossalai/nn/layer/moe/_operation.py [hunk 2]
```diff
@@ -70,20 +67,20 @@ class AllToAll(torch.autograd.Function):
     """
 
     @staticmethod
-    def forward(ctx: Any, inputs: Tensor, parallel_mode: ParallelMode) -> Tensor:
+    def forward(ctx: Any, inputs: Tensor, group=None) -> Tensor:
```
Comment (feifeibear) [inline colossalai/nn/layer/moe/_operation.py:70]: missed typing hit

File: colossalai/nn/layer/moe/_operation.py (hunk 1 above)
Comment (1SAA) [reply colossalai/nn/layer/moe/_operation.py]: done

File: colossalai/nn/layer/moe/_operation.py (hunk 2 above)
Comment (1SAA) [reply colossalai/nn/layer/moe/_operation.py]: done

File: colossalai/nn/layer/moe/_operation.py [hunk 3]
```diff
@@ -18,35 +16,35 @@
 class AllGather(torch.autograd.Function):
 
     @staticmethod
-    def forward(ctx: Any, inputs: Tensor, parallel_mode: ParallelMode) -> Tensor:
+    def forward(ctx: Any, inputs: Tensor, group: ProcessGroup = None) -> Tensor:
```
Comment (feifeibear) [inline colossalai/nn/layer/moe/_operation.py:19]: Optional[ProcessGroup]

File: colossalai/nn/layer/moe/_operation.py [hunk 4]
```diff
@@ -70,20 +68,20 @@ class AllToAll(torch.autograd.Function):
     """
 
     @staticmethod
-    def forward(ctx: Any, inputs: Tensor, parallel_mode: ParallelMode) -> Tensor:
+    def forward(ctx: Any, inputs: Tensor, group: ProcessGroup = None) -> Tensor:
```
Comment (feifeibear) [inline colossalai/nn/layer/moe/_operation.py:71]: Optional[ProcessGroup]

**Review by feifeibear — APPROVED** [review]

**Pull request merged**